presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -388,6 +389,11 @@ public SystemSessionProperties(
                         FILTER_AND_PROJECT_MIN_OUTPUT_PAGE_ROW_COUNT,
                         "Experimental: Minimum output page row count for filter and project operators",
                         featuresConfig.getFilterAndProjectMinOutputPageRowCount(),
+                        false),
+                booleanSessionProperty(
+                        USE_MARK_DISTINCT,
```

Comment:
can you provide some guideline when using `MarkDistinctNode` is better or worse?

Reply:
I still need to test this, but my intuition is that it should help with queries that contain many DISTINCT aggregates with different inputs and other aggregates on many columns, since it avoids multiple shuffles of the entire data set.

It can be worse if the inputs to DISTINCT aggregates have high cardinality, since the "mark distinct" part won't execute in a distributed way. It could cause those queries to fail with "local memory exceeded"